File: 8_kyu/Find_the_smallest_integer_in_the_array.lua
-- Functions returns the minimum of the items in the table t.
return function(t)
  if not t or #t == 0 then
    return
  end;
  local minimum = t[1]
  for i =1,#t do
    if t[i] < minimum then
      minimum = t[i]
    end
  end
  return minimum
end

File: 8_kyu/Find_the_smallest_integer_in_the_array.c
#include <stddef.h>

/**
 *  Returns smallest integer of array.
 *  array  : Not empty array of integers
 *  len    : length of array
 *  return : minimum of array
 */
int find_smallest_int(const int array[/* len */], size_t len)
{
  // error exit on empty or no array
  if(len == 0 || array == NULL)
    return 0;
  // smallest integer seen so far
  int m = array[0];
  for( size_t i =0; i < len; ++i)
    if(array[i] < m ) m = array[i];
  return m;
}